crates/nu-cmd-plugin/src/commands/plugin/add.rs

```diff
@@ -0,0 +1,148 @@
+use std::sync::Arc;
+
+use nu_engine::{command_prelude::*, current_dir};
+use nu_plugin::{GetPlugin, PersistentPlugin};
+use nu_protocol::{PluginCacheItem, PluginGcConfig, PluginIdentity, RegisteredPlugin};
+
+use crate::util::modify_plugin_file;
+
+#[derive(Clone)]
+pub struct PluginAdd;
+
+impl Command for PluginAdd {
+    fn name(&self) -> &str {
+        "plugin add"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build(self.name())
+            .input_output_type(Type::Nothing, Type::Nothing)
+            // This matches the option to `nu`
+            .named(
+                "plugin-config",
+                SyntaxShape::Filepath,
+                "Use a plugin signature file other than the one set in `$nu.plugin-path`",
+                None,
+            )
+            .named(
+                "shell",
+                SyntaxShape::Filepath,
+                "Use an additional shell program (cmd, sh, python, etc.) to run the plugin",
+                Some('s'),
+            )
+            .required(
+                "filename",
+                SyntaxShape::Filepath,
+                "Path to the executable for the plugin",
+            )
+            .category(Category::Plugin)
+    }
+
+    fn usage(&self) -> &str {
+        "Add a plugin to the plugin signature file."
+    }
+
+    fn extra_usage(&self) -> &str {
+        r#"
+This does not load the plugin commands into the scope - see `register` for that.
+
+Instead, it runs the plugin to get its command signatures, and then edits the
+plugin signature file (by default, `$nu.plugin-path`). The changes will be
+apparent the next time `nu` is next launched with that plugin signature file.
+"#
+        .trim()
+    }
+
+    fn search_terms(&self) -> Vec<&str> {
+        vec!["plugin", "add", "register", "load", "signature"]
+    }
+
+    fn examples(&self) -> Vec<Example> {
+        vec![
+            Example {
+                example: "plugin add nu_plugin_inc",
+                description: "Run the `nu_plugin_inc` plugin from the current directory or $env.NU_PLUGIN_DIRS and install its signatures.",
+                result: None,
+            },
+            Example {
+                example: "plugin add --plugin-config=polars.msgpackz nu_plugin_polars",
```

Comment:
I'm not a super fan of this syntax `--plugin-config=polars.msgpackz`. Can we do the same thing but without the equal sign?